.github/DESIGN.md

```diff
@@ -15,6 +26,32 @@ The user understands the basic mechanics of a CLI:
 - CTRL + C quits any running task
 - how to navigate and manipulate the filesystem via the command line: cd and mkdir
 
+## Components of a well designed command
+When creating a new command or subcommand there are a few things to keep in mind.
+
+### Reduce user input
+When trying to create parity with an existing GUI based feature, consider the larger task at hand and build your command to accomplish that goal. 
+
+For example when creating a `dev-store` the data model requires up to 8 fields to be filled in. When creating a CLI command to create a `dev-store`, don't ask the user for all 8 fields - auto-fill all 8 fields with smart generated data. If a field requires an address dont generate an animal name, try to get as close as possible to the format the data expects.
```

Comment:
I fixed a couple typos below, but I think there's a more fundamental question to think about here.

To me, `auto-fill all 8 fields` is not necessarily good advice to give. The data model exists for a reason — if it doesn't matter what goes in there, we should be re-evaluating the data model more fundamentally, not filling it with dummy text. I think the guidance should instead be: ask only for as much input as is necessary to complete the task in question. And some of that input genuinely matters!

> For example, when creating a `dev-store`, the data model requires up to 8 fields to be filled in. When creating a CLI command to create a `dev-store`, don't ask the user for all 8 fields — auto-fill all 8 fields with smart generated data. If a field requires an address, don’t generate an animal name; try to get as close as possible to the format the data expects.

Reply:
I agree but I would challenge that. Im not suggesting we change the fundamental data model, like you said it exists for a reason. Im suggesting how you get around the data model. For dev stores specifically the data model exists to ensure Merchants fill in the correct info, and Partners have inherited that - the goal for a dev is to get a store going as fast as possible so the actual. data needed in all those fields is far less important than creating a functional store.

Reply:
I get where you're coming from, but the phrase `get around the data model` is a red flag for me, and a sign that we have to consider what behaviors will result from this advice. I think it encourages potentially troublesome design choices in the long run!

Reply:
As a discussion point, isn't this the way we're handling all data generation for things like products, orders, and customers? We recognize the individual product or customer is not important so we can generate data, but the data model requires a minimum amount of information to allow the objects to be created successfully.

Ultimately I think this may actually be a bad example for external teams contributing to the CLI will not be working with the Partners API or data model. I think the principle is still sound, but I'd be ok going with a different example.

Reply:
I removed the reference to `dev-store` and went with a simpler command and value here instead. I think its a fair point, and I don't know if devs will need to work with Partner Dash so I side stepped the whole data model and how to get around it instead. How does the updated content feel?

Reply:
👍 Looks good to me!

Couple little typos: **execuexecuted**, **Argumemnts**